string date
float64[] best_query
float64 outcome